src/objective/rank_objective.hpp

```diff
@@ -135,19 +146,34 @@ class LambdarankNDCG : public RankingObjective {
     }
     // construct Sigmoid table to speed up Sigmoid transform
     ConstructSigmoidTable();
+
+    // initialize position bias vectors
+    InitPositionBiasesAndGradients();
```

Comment:
should this initialization be conditional on the value of `unbiased_`? Otherwise, the default behavior will be to initialize them and then not use them, right?

(apologies if I've misunderstood how this works)

Reply:
they get initialized to 1 and only updated if `unbiased_` is true. so any time you see them when `unbiased_` is false, you're just dividing by 1.